rasterio-stubs/io.pyi

```diff
@@ -26,9 +26,13 @@ from rasterio.windows import WindowMethodsMixin
 
 class DatasetReader(DatasetReaderBase, WindowMethodsMixin, TransformMethodsMixin):
     def __repr__(self) -> str: ...
+    def __enter__(self, *args, **kwargs) -> DatasetReader: ...
+    def __exit__(self, *args, **kwargs): ...
```

Comment:
These don't exist in the rasterio implementation of `DatasetReader`: https://github.com/rasterio/rasterio/blob/38ad052cc2ab460d0b632bd2f3d7919e3d96c3c8/rasterio/io.py#L28-L34

Ideally they'd be put on `DatasetBase`:
https://github.com/rasterio/rasterio/blob/38ad052cc2ab460d0b632bd2f3d7919e3d96c3c8/rasterio/_base.pyx#L436-L441